edx_proctoring/backends/software_secure.py

```diff
@@ -129,13 +120,26 @@ def get_software_download_url(self):
         """
         return self.software_download_url
 
-    def on_review_callback(self, payload):
+    def on_review_callback(self, attempt, payload):
         """
         Called when the reviewing 3rd party service posts back the results
 
         Documentation on the data format can be found from SoftwareSecure's
         documentation named "Reviewer Data Transfer"
         """
+        received_id = payload['examMetaData']['ssiRecordLocator'].lower()
+        if attempt['external_id'].lower() != received_id and not\
+                settings.PROCTORING_SETTINGS.get('ALLOW_CALLBACK_SIMULATION', False):
```

Comment:
I found the original form of this test easier to understand (fewer negations, and split out into a variable):
```
        match = (
            attempt_obj.external_id.lower() == external_id.lower() or
            settings.PROCTORING_SETTINGS.get('ALLOW_CALLBACK_SIMULATION', False)
        )
        if not match:
```